### ISSUE TITLE
mpl circuit drawer fails for empty circuit

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-terra#2097]
### Information

- **Qiskit Terra version**: 0.7.1
- **Python version**:
- **Operating system**:

### What is the current behavior?
```python
qreg = QuantumRegister(5, 'qreg')
creg = ClassicalRegister(5, 'creg')
qc = QuantumCircuit(qreg, creg)
qc.draw(output='mpl')
```

```python

---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-9-2250cfe3db72> in <module>
      2 creg = ClassicalRegister(5, 'creg')
      3 qc = QuantumCircuit(qreg, creg)
----> 4 qc.draw(output='mpl')

~/miniconda3/lib/python3.7/site-packages/qiskit/circuit/quantumcircuit.py in draw(self, scale, filename, style, output, interactive, line_length, plot_barriers, reverse_bits)
    351                                             line_length=line_length,
    352                                             plot_barriers=plot_barriers,
--> 353                                             reverse_bits=reverse_bits)
    354 
    355     def size(self):

~/miniconda3/lib/python3.7/site-packages/qiskit/tools/visualization/_circuit_visualization.py in circuit_drawer(circuit, scale, filename, style, output, interactive, line_length, plot_barriers, reverse_bits)
    235                                                filename=filename, style=style,
    236                                                plot_barriers=plot_barriers,
--> 237                                                reverse_bits=reverse_bits)
    238         else:
    239             raise _error.VisualizationError(

~/miniconda3/lib/python3.7/site-packages/qiskit/tools/visualization/_circuit_visualization.py in _matplotlib_circuit_drawer(circuit, scale, filename, style, plot_barriers, reverse_bits)
    575                                        reverse_bits=reverse_bits)
    576     qcd.parse_circuit(circuit)
--> 577     return qcd.draw(filename)

~/miniconda3/lib/python3.7/site-packages/qiskit/tools/visualization/_matplotlib.py in draw(self, filename, verbose)
    341     def draw(self, filename=None, verbose=False):
    342         self._draw_regs()
--> 343         self._draw_ops(verbose)
    344         _xl = - self._style.margin[0]
    345         _xr = self._cond['xmax'] + self._style.margin[1]

~/miniconda3/lib/python3.7/site-packages/qiskit/tools/visualization/_matplotlib.py in _draw_ops(self, verbose)
    464         _barriers = {'coord': [], 'group': []}
    465         next_ops = self._ops.copy()
--> 466         next_ops.pop(0)
    467         this_anc = 0
    468 

IndexError: pop from empty list


```

**Comment by mtreinish** [comment]
This should have been fixed by #1866. Can you check if you still have this issue on master? I can push up a patch adding a test for this (since no output and no exception is easy to test against) to verify it doesn't regress in the future.

**Comment by nonhermitian** [comment]
Ok cool.  I am currently travelling and making slides from `0.7` so do not have time to check.  But it does look like that PR solves the problem, so feel free to close.
